internal/e2etests/rdns/resource_test.go

```diff
@@ -97,6 +97,71 @@ func TestRDNSResource_Server(t *testing.T) {
 	}
 }
 
+func TestRDNSResource_PrimaryIP(t *testing.T) {
+	tests := []struct {
+		name          string
+		dns           string
+		primaryIPType string
+	}{
+		{
+			name:          "primary-ipv6-rdns",
+			dns:           "ipv6.example.hetzner.cloud",
+			primaryIPType: "ipv6",
+		},
+		{
+			name:          "primary-ipv4-rdns",
+			dns:           "ipv4.example.hetzner.cloud",
+			primaryIPType: "ipv4",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var primaryIP hcloud.PrimaryIP
+
+			tmplMan := testtemplate.Manager{}
+			restPrimaryIP := &primaryip.RData{
+				Name:         tt.name,
+				Type:         tt.primaryIPType,
+				AssigneeType: "server",
+				Datacenter:   e2etests.TestDataCenter,
+			}
+			restPrimaryIP.SetRName(tt.name)
+			resRDNS := rdns.NewRDataPrimaryIP(t, tt.name, restPrimaryIP.TFID()+".id", restPrimaryIP.TFID()+".ip_address", tt.dns)
+
+			// TODO: Debug issues that causes this to fail when running in parallel
```

Comment:
Do we want to let this TODO make it into `main`?

Reply:
Its already in main since #660 for all other tests in this file:

https://github.com/hetznercloud/terraform-provider-hcloud/blob/1bf4bae0d5bd9d64adada24828e6062bee564118/internal/e2etests/rdns/resource_test.go#L66